src/components/Bank.js

```diff
@@ -105,14 +105,29 @@ class Bank extends Component {
     return filtered;
   };
 
+  // buy = () => {
+  //   const { userBalances, base, amount } = this.state;
+  //   const destinationBalance = userBalances.filter(x => x.type === base)[0];
+  //   console.log(destinationBalance);
+
+  //   if (destinationBalance) {
+  //     this.setState({
+  //       userBalances: (userBalances[destinationBalance] = base)
+  //     });
+  //   } else {
+  //     this.setState({
+  //       userBalances: userBalances.push({ type: base, value: Number(amount) })
+  //     });
+  //   }
+  // };
   buy = e => {
```

Comment:
I think this thing is pretty tricky: 

It has to do a couple of things

- add a currency if it's not present
- add money to one currency
- subtract it from another one
- It also checks if you have an amount and if you're not overdrawing your account

I've divided the code that wasn't working code into two steps. 
1.) add a currency if not present (if else statement)
2.) add and subtract money to different currencies. I use array.map since the amount of entries in the array stays the same, but the data has to update. 

```javascript
  buy = e => {
    e.preventDefault();
    const { amount, userBalances, exchangeTo, base, result } = this.state;

    if (amount === isNaN) {
      return;
    }
    // console.log(userBalances.filter(x => x.type === exchangeTo)[0].value);
    if (result > userBalances.find(x => x.type === exchangeTo).value) {
      alert('Insufficent Balance');
      return;
    }
    if (amount <= 0) {
      alert('Please Enter Positive Number');
      return;
    }

    // I made this thing into a function, we could take it out to different file if needed.
    const updateBalances = (currentBalances, baseCurrency, exchangeToCurrency, amount, result) => {
      const destinationBalance = currentBalances.find(x => x.type === baseCurrency);
      let balancesToUpdate

      // 1.)
      // If we don 't have the destination currency, add it and start it at 0
      if(!destinationBalance){
        balancesToUpdate = [...currentBalances, {
          type: baseCurrency,
          value: 0
        }]
      } else {
        // if we already have the the currencies involved, we don't do anything
        balancesToUpdate = currentBalances
      }
      
      // 2.)
      return balancesToUpdate.map(balance => {
        // this balance is the base currency, update object by adding the amount
        if (balance.type === baseCurrency) {
          return {
            type: baseCurrency,
            value: Number(amount) + balance.value
          };
        }

        // this balance is the exchangeToCurrenct, update the object by subtracting the amount
        if (balance.type === exchangeToCurrency) {
          return {
            type: exchangeToCurrency,
            value:
              balance.value - Number(result)
          };
        }

        // this is a different currency that has nothing to do with this transaction
        // do nothing
        return balance
      });
    }

    this.setState({
      userBalances: updateBalances(userBalances, base, exchangeTo, amount, result)
    });
  };
```